spec/acceptance/class_spec.rb

```diff
@@ -1,62 +1,141 @@
+  #     token                  => 'abcd1234',
+  #     record_event           => true,
+  #     pe_username            => 'admin',
+  #     pe_console             => 'localhost',
+  #     pe_password            => Sensitive('pie'),
+  #     include_api_collection => true,
+  #   }
+  #   MANIFEST
+
+  #   it 'Sets up the pe-puppetserver service and splunk_hec class' do
+  #     apply_manifest(cleanmanifest, catch_failures: true)
+  #   end
+
+  #   it 'Successfully creates a report after a simple puppet apply' do
+  #     host = return_host
+  #     run_shell('puppet apply -e \' notify { "Hello World" : }\' --reports=splunk_hec')
+  #     expect(run_shell("ls /opt/puppetlabs/puppet/cache/reports/#{host}").stdout).to match %r{\.yaml}
+  #   end
+
+  #   it 'Successfully sends data to an http endpoint' do
+  #     run_shell('cat /etc/puppetlabs/code/environments/production/modules/splunk_hec/examples/orchestrator_metrics.json | puppet splunk_hec --sourcetype puppet:summary --saved_report')
+  #   end
+
+  #   it 'Fails when missing required parameters' do
+  #     failmanifest = <<-MANIFEST
+  #       service { 'pe-puppetserver':
+  #         ensure     => running,
+  #         hasrestart => true,
+  #         restart    => "service pe-puppetserver reload",
+  #       }
+  #       class { 'splunk_hec':
+  #         url => 'notanendpoint/nicetry',
+  #         token => '',
+  #         record_event => true,
+  #       }
+  #       MANIFEST
+  #     expect { apply_manifest(failmanifest, catch_failures: true) }.to raise_error(RuntimeError)
+  #   end
+
+  #   it 'Fails when given a bad endpoint' do
+  #     failmanifest = <<-MANIFEST
+  #       service { 'pe-puppetserver':
+  #         ensure     => running,
+  #         hasrestart => true,
+  #         restart    => "service pe-puppetserver reload",
+  #       }
+  #       class { 'splunk_hec':
+  #         url                    => 'notanendpoint/nicetry',
+  #         token                  => '',
+  #         record_event           => true,
+  #         pe_username            => 'admin',
+  #         pe_console             => 'localhost',
+  #         pe_password            => Sensitive('pie'),
+  #         include_api_collection => true,
+  #       }
+  #       MANIFEST
+  #     apply_manifest(failmanifest, catch_failures: true)
+  #     cmd = 'cat /etc/puppetlabs/code/environments/production/modules/splunk_hec/examples/foo.json | puppet splunk_hec --sourcetype puppet:summary --saved_report'
+  #     results = run_shell(cmd, expect_failures: true).to_s
+  #     expect(results).to match %r{exit_code=1}
+  #   end
+  # end
 
-    it 'Successfully sends data to an http endpoint' do
-      run_shell('cat /etc/puppetlabs/code/environments/production/modules/splunk_hec/examples/orchestrator_metrics.json | puppet splunk_hec --sourcetype puppet:summary --saved_report')
+  it 'Collect events from PE and sends to the splunk endpoint' do
```

Comment:
Correct me if I'm wrong, but this entire test is really just setup for the next test. This makes the next test dependent on this one which is something we want to avoid in RSPEC. You should be able to invoke any single test in the file, and the dependencies for that test should be mapped out via things like `before (:all)` and `before (:each)` kinds of blocks that it will run successfully.

You could replace this test with a smaller commandline command that can be placed directory in a `before (:each)` block or perhaps in an `include context` command. The commandline to run is as follows:
`for i in {1..4}; do puppet task run enterprise_tasks::test_connect -n abortive-magic.delivery.puppetlabs.net; done`.
Put that string in a `run_cmd` litmus command, and it will run against the master and those tasks will be invoked via the orchestrator and the script will pick them up as events to send to splunk. If you want to send events for Puppet job runs instead of tasks and plans you can invoke `puppet job run -n abortive-magic.delivery.puppetlabs.net` each time through the loop instead, and those will be job runs that the script will pick up and send to splunk.

Reply:
Gotcha

Reply:
To get the fqdn of the server you're running the Bolt command on isn't as straightforward as I would have liked it to be, but I think you can run the following command inside `run_cmd` and it will work:
```
puppet job run -n `hostname -A | cut -d " " -f1`
```